### ISSUE TITLE
[HOLD for payment 2023-08-16] [$1000] Payments - On add debit card page state field seems to have extra horizontal spacing

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#23869]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Login to ND 
2. Press Avatar button to open settings page
3. Navigate to Payments ->  Add Payment method ->Select Debit card
4. Notice State field has extra margin horizontal than the other inputs

## Expected Result:
State field should not have extra horizontal spacing like text input

## Actual Result:
 Extra horizontal spacing on state field

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.47-5
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

![Simulator Screenshot - iPhone 14 - 2023-07-29 at 02 26 12](https://github.com/Expensify/App/assets/93399543/d4384db3-cebd-4ffd-90a1-d8bf97a88ceb)

![image](https://github.com/Expensify/App/assets/93399543/123acf9f-e19c-47e5-8503-1a9530eff53c)

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhairyasenjaliya
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1690577968526579

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0199c0e50fda86a4c5</li>
        <li>Upwork Job ID: 1686055085493682176</li>
        <li>Last Price Increase: 2023-07-31</li>
<li>Automatic offers: </li>
<ul>
<li>dhairyasenjaliya | Contributor | 25928715</li>
<li>dhairyasenjaliya | Reporter | 25928718</li>
</ul></li>
    </ul>
</details>

**Comment by kbecciv** [comment]
### Proposal  

### Please re-state the problem that we are trying to solve in this issue. 

- On add debit card page state field seems to have extra horizontal spacing

### What is the root cause of that problem?

- The root cause is we on `<Form>` we have added `styles.mh5` that will add horizontal spacing to entire page but for state picker we have not adjusted horizontal padding just like other pages that will have opacity to entire `<Menu>` just like other pages 

### What changes do you think we should make in order to solve the problem?

### Solution - 1

- On `AddDebitCardPage.js` page we just need to add `styles.mhn5` to the `<View>` that basically wraps the `<StatePicker>`
- We are using same `styles.mhn5` to other pages we have added  `<StatePicker>`

### Solution - 2

- For `<StatePicker>` component we can define new `props` to allow `styles` from parent 
- And then we can just pass style of `marginTop` and `marginHorizontal`  to `<StatePicker>` 
- We can remove `<View>` that wraps component from all the pages that we have used `<StatePicker>` E.g : `AddressForm.js`, `CompanyStep.js`, `AddDebitCardPage.js`, `AddressPage.js` 
- It will be cleanup to remove `<View>` that will be not necessary after providing `styles` directly to `component`

### Solution - 3

- We can add `styles.mhn5` directly to  `<StatePicker>` 

### What alternative solutions did you explore? (Optional)

-  N/A
———————

**Comment by garrettmknight** [comment]
Easy, Melv. We're waiting for proposals.

**Comment by sobitneupane** [comment]
Solution 1 from @dhairyasenjaliya [proposal](https://github.com/Expensify/App/issues/23869#issuecomment-1656805944) looks good to me. We are using the same solution in multiple places.

🎀 👀 🎀 C+ reviewed

**Comment by garrettmknight** [comment]
@dhairyasenjaliya offer sent!

**Comment by garrettmknight** [comment]
**Summarizing payments for this issue:**

#urgency bonus? Yes
Reporter: @dhairyasenjaliya  $250 paid via Upwork
Contributor: @dhairyasenjaliya  $1500 paid via Upwork
Contributor+: @sobitneupane  $1500

Upwork job is here: https://www.upwork.com/jobs/~0199c0e50fda86a4c5

**Comment by garrettmknight** [comment]
@sobitneupane I'll pay you out when you finish the checklist.

**Comment by sobitneupane** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
> 
> * [ ]  [@sobitneupane] The PR that introduced the bug has been identified. Link to the PR:

https://github.com/Expensify/App/pull/21049

> * [ ]  [@sobitneupane] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

https://github.com/Expensify/App/pull/21049#issuecomment-1685730240

> * [ ]  [@sobitneupane] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

This is already covered by PR review checklist.

> * [ ]  [@sobitneupane] Determine if we should create a regression test for this bug.

I don't think a separate regression test is required for this bug.

**Comment by garrettmknight** [comment]
Assigned @JmillsExpensify for payment.

**Comment by garrettmknight** [comment]
I think this is just awaiting payment from @JmillsExpensify

**Comment by JmillsExpensify** [comment]
I'll jump in today for all NewDot payments. Feel free to close this in the meantime.

**Comment by JmillsExpensify** [comment]
Reviewed the details for @sobitneupane. $1,500 payment approved via NewDot based on BZ summary.

**Comment by amyevans** [comment]
Looks like we're all set here!
